WebDriverAgentLib/Categories/XCUIElement+FBUtilities.m

```diff
@@ -65,11 +66,24 @@ - (BOOL)fb_obstructsElement:(XCUIElement *)element
 
 - (XCElementSnapshot *)fb_lastSnapshot
 {
-  [self query];
-  [self resolve];
+  [self fb_resolve];
   return self.lastSnapshot;
 }
 
+- (void)fb_resolve
+{
+  if ([self isKindOfClass:XCUIApplication.class]) {
+    [((XCUIApplication *)self) fb_resolve];
+  }
+  else if (self.application.fb_isActivateSupported) {
```

Comment:
this `if` is redundant

Reply:
There are three cases here:

1. Self is XCUIApplication and supports activate call fb_resolve
2. If Activate is supported (and it is not XCUIApplication) get access the application property, call fb_resolve
3. Activate is not supported, do as before

Reply:
@mykola-mokhnach I now see what you mean, I was missing the fb_isActivateSupported in the first if statement

Reply:
The code _should_ be

```
if ([self isKindOfClass:XCUIApplication.class] && self.application.fb_isActivateSupported) {
    [((XCUIApplication *)self) fb_resolve];
  } else if (self.application.fb_isActivateSupported) {
    [self.application fb_resolve];
  } else {
    [self query];
    [self resolve];
  }
```

Reply:
@mykola-mokhnach simplified the code again to

```
  if (self.application.fb_isActivateSupported) {
    [self.application fb_resolve];
  } else {
    [self query];
    [self resolve];
  }
```

There was no need to cast self to XCUIApplication as there a property available on self already